### ISSUE TITLE
Feature request: custom indentation setting

### ISSUE THREAD

**Issue by rvnkcode** [heyman/heynote#245]
**Have you searched existing Github issues to see if someone has already requested this feature?**
Yes

**Is your feature request related to a problem? Please describe.**
It would be nice to be able to customize the indentation settings when formatting code. Like the use of spaces/tabs or 2/4 spaces.

**Describe the solution you'd like**
Add a formatting menu to the settings so users can set whether to use spaces or tabs when formatting, and how many spaces to use. It would be great to be able to set it per programming language.

**Describe alternatives you've considered**
Change the indentation setting for JS/TS/Markdown/JSON to 2 spaces.

**Comment by junyeongh** [comment]
Agreed. It would be nice if there is a vscode-like bottom bar indentation menu.

[vscode/src/vs/editor/contrib/indentation at main · microsoft/vscode](https://github.com/microsoft/vscode/tree/main/src/vs/editor/contrib/indentation)

**Comment by tmchow** [comment]
Agreed, I really want 2 spaces vs the current 4.

**Comment by heyman** [comment]
Fixed by #311 and #327
